src/model.cpp

```diff
@@ -99,10 +129,36 @@ CalcReportBundle Model::generateVolumeData(){
   auto end = std::chrono::steady_clock::now();
   _data.addTime(std::chrono::duration<double>(end-start).count());
 
-  // create a report bundle
   return calcVolume();
 }
 
+CalcReportBundle Model::generateSurfaceData(){
+  // TODO: add way for this function to tell whether volume calculation has been conducted before
+  auto start = std::chrono::steady_clock::now();
+
+  std::vector<std::vector<char>> solid_types = 
+  { {0b00000011},
+    {0b00000011, 0b00000101},
+    {0b00000011, 0b00000101, 0b00100001, 0b01000001},
+    {0b00000011, 0b00000101, 0b00010001, 0b00100001, 0b01000001}};
+
+  std::vector<bool> for_every_cavity = {false, false, true, true};
+
+  std::vector<std::vector<double>> surface_areas = _cell.sumSurfArea(solid_types, for_every_cavity, _data.cavities.size());
+
+  _data.surf_vdw = surface_areas[0][0]; // 0b00000011
+  _data.surf_probe_inaccessible = surface_areas[1][0]; // 0b00000011 + 0b00000101

```

Comment:
The total probe 1 accessible surface area will be missing with the current branch.
Need to consider changes to add `_data.surf_probe1_accessible = surface_areas[3][0];` for total in addition of per cavity.

Reply:
Consider opening this as an issue, so we don't forget. What I wanted to note however, is that I think it's best to avoid using `probe1` or `probe2` in the code, because understanding them requires deeper knowledge of how everything works. It's less descriptive than for instance `probe_small` or `probe_large`, or something similar that includes the defining property in the name.